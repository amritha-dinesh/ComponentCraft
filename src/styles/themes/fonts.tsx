import { Platform, TextStyle } from "react-native";

export const typeface = {
  fontFamily: Platform.select({
    web: "Roboto Mono, monospace",
    ios: "System",
    android: "Roboto Mono, monospace, sans-serif",

    default: "sans-serif",
  }),

  weightRegular: "400" as const,

  weightMedium: "500" as const,

  weightBold: "700" as const,
};

export const regularType: TextStyle = {
  fontFamily: typeface?.fontFamily,

  fontWeight: typeface?.weightRegular,

  letterSpacing: 0,
};

export const boldType: TextStyle = {
  fontFamily: typeface?.fontFamily,

  fontWeight: typeface?.weightBold,

  letterSpacing: 0.15,
};
