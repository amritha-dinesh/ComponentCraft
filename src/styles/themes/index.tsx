export { spacing } from "./spacing";
export { theme } from "./theme";
export { colors } from "./colors";
export { componentSizes } from "./constants";
export { typeface, regularType, boldType } from "./fonts";
