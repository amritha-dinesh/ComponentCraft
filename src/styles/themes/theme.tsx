import { DarkTheme, DefaultTheme } from "@react-navigation/native";
import { lightThemeColors, darkThemeColors } from "./colors";
import { spacing } from "./spacing";
import { componentSizes } from "./constants";

export const defaultFontFamily = {
  web: "Roboto Mono, monospace",
  ios: "System",
  android: "Roboto Mono, monospace, sans-serif",
  default: "sans-serif",
};

export const theme = {
  light: {
    ...DefaultTheme,
    colors: {
      ...DefaultTheme.colors,
      ...lightThemeColors,
    },
    fonts: defaultFontFamily,
    spacing,
    componentSizes,
  },
  dark: {
    ...DarkTheme,
    colors: {
      ...DarkTheme.colors,
      ...darkThemeColors,
    },
    fonts: defaultFontFamily,
    spacing,
    componentSizes,
  },
};

export const darkTheme = theme.dark;
export const lightTheme = theme.light;
