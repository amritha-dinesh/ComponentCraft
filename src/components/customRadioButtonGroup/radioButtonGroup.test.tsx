import React from "react";
import { render, fireEvent } from "@testing-library/react-native";
import RadioButtonGroup from "./radioButtonGroup";
import { green500 } from "../../styles/themes/colors";

describe("CustomCheckboxGroup", () => {
  it("should match the snapshot", () => {
    const options = [
      { label: "Java", value: "Java" },
      { label: "Kotlin", value: "Kotlin" },
      { label: "Javascript", value: "Javascript" },
    ];
    const { toJSON } = render(<RadioButtonGroup options={options} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it("renders correctly", () => {
    const { getByTestId } = render(<RadioButtonGroup options={[]} />);
    const customCheckboxGroup = getByTestId("radio-button-group");
    expect(customCheckboxGroup).toBeDefined();
  });

  it("should render the correct number of radio buttons", () => {
    const options = [
      { label: "Java", value: "Java" },
      { label: "Kotlin", value: "Kotlin" },
      { label: "Javascript", value: "Javascript" },
    ];
    const { getAllByTestId } = render(<RadioButtonGroup options={options} />);
    const radioButton = getAllByTestId("radio-button");
    expect(radioButton.length).toBe(options.length);
  });

  it("should select the correct option when clicked", () => {
    const options = [
      { label: "Option 1", value: "Option 1" },
      { label: "Option 2", value: "Option 2" },
      { label: "Option 3", value: "Option 3" },
    ];
    const { getByText } = render(
      <RadioButtonGroup options={options} labelColor={green500} />
    );
    fireEvent.press(getByText("Option 2"));
    const selectedOption = getByText("Option 2");
    const textStyle = selectedOption.props.style;
    const colorStyle = Array.isArray(textStyle)
      ? textStyle.find((style) => style.hasOwnProperty("color"))
      : textStyle;
    const color = colorStyle.color;

    expect(color).toBe(green500);
  });

  it("should call the onSubmit callback with the selected option", () => {
    const options = [
      { label: "Option 1", value: "Option 1" },
      { label: "Option 2", value: "Option 2" },
      { label: "Option 3", value: "Option 3" },
    ];
    const onSubmitMock = jest.fn();
    const { getByText } = render(
      <RadioButtonGroup options={options} onSubmit={onSubmitMock} />
    );
    fireEvent.press(getByText("Option 3"));
    expect(onSubmitMock).toHaveBeenCalledWith({
      label: "Option 3",
      value: "Option 3",
    });
  });
});
