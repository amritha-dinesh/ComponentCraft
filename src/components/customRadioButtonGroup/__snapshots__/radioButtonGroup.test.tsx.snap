// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomCheckboxGroup should match the snapshot 1`] = `
<RCTSafeAreaView
  style={
    {
      "alignSelf": "flex-start",
    }
  }
  testID="radio-button-group"
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "flexDirection": "row-reverse",
        "justifyContent": "space-between",
        "marginBottom": 8,
        "opacity": 1,
      }
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "borderColor": "#000",
            "borderRadius": 12,
            "borderWidth": 2,
            "height": 24,
            "justifyContent": "center",
            "width": 24,
          },
        ]
      }
      testID="radio-button"
    />
    <Text
      style={
        [
          {
            "color": "#000000",
            "fontFamily": undefined,
            "fontSize": undefined,
            "marginLeft": 0,
            "marginRight": 10,
          },
        ]
      }
    >
      Java
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "flexDirection": "row-reverse",
        "justifyContent": "space-between",
        "marginBottom": 8,
        "opacity": 1,
      }
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "borderColor": "#000",
            "borderRadius": 12,
            "borderWidth": 2,
            "height": 24,
            "justifyContent": "center",
            "width": 24,
          },
        ]
      }
      testID="radio-button"
    />
    <Text
      style={
        [
          {
            "color": "#000000",
            "fontFamily": undefined,
            "fontSize": undefined,
            "marginLeft": 0,
            "marginRight": 10,
          },
        ]
      }
    >
      Kotlin
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "flexDirection": "row-reverse",
        "justifyContent": "space-between",
        "marginBottom": 8,
        "opacity": 1,
      }
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "borderColor": "#000",
            "borderRadius": 12,
            "borderWidth": 2,
            "height": 24,
            "justifyContent": "center",
            "width": 24,
          },
        ]
      }
      testID="radio-button"
    />
    <Text
      style={
        [
          {
            "color": "#000000",
            "fontFamily": undefined,
            "fontSize": undefined,
            "marginLeft": 0,
            "marginRight": 10,
          },
        ]
      }
    >
      Javascript
    </Text>
  </View>
</RCTSafeAreaView>
`;
